src/gordon_gcp/clients/auth.py

```diff
@@ -178,14 +178,38 @@ async def refresh_token(self):
                 specifically a :exc:`.GCPHTTPResponseError`, if the
                 exception is associated with a response status code.
         """
+        if isinstance(
+                self.creds, compute_engine.credentials.Credentials):
```

Comment:
In general, it is good practice to try to minimize use of `isinstance` (some adage similar to "check actions, not types"). It also seems like this probably isn't something that will change over the lifetime of the client, could the appropriate function could be set while setting up the credentials from the init method rather than on every refresh request?

Reply:
Yep, I am aware that `isinstance` is not best practice in python. I could check on the existence of all the fields like `client_id`, `client_secret` and stuff, but even then the real differentiating factor is where the credentials are from (service account file, compute engine, etc) and that information is just in the type as far as I can see.
I can move the check in the init function and then just assign the function to some field and call it here.